lib/etcd_databag.py

```diff
@@ -70,14 +70,11 @@ def __init__(self):
 
     def cluster_token(self):
         ''' Getter to return the unique cluster token. '''
-        if not is_leader():
-            return leader_get('token')
-
-        if not self.db.get('cluster-token'):
+        token = leader_get('token')
```

Comment:
leader_get('*token*') -> leader_set({'*cluster-token*': token})

Is that correct? It seems like this confusion predates this change.

Reply:
The heck? Good catch, thanks. I need to figure out how the hell this didn't blow up in testing.

Reply:
Yep, the followers were not using the correct token:

```
$ juju run --application etcd -- cat /var/snap/etcd/common/etcd.conf.yml '|' grep token
- Stdout: |
    # Initial cluster token for the etcd cluster during bootstrap.
    initial-cluster-token: 'ZYFRQ6'
  UnitId: etcd/0
- Stdout: |
    # Initial cluster token for the etcd cluster during bootstrap.
    initial-cluster-token: 'None'
  UnitId: etcd/1
- Stdout: |
    # Initial cluster token for the etcd cluster during bootstrap.
    initial-cluster-token: 'None'
  UnitId: etcd/2
```

Yet somehow the cluster came up fine anyway. Seems like the tokens aren't actually being used. Weird.

In any case, I fixed the `leader_set` and tested a new deployment. This time, the tokens look good:

```
$ juju run --application etcd -- cat /var/snap/etcd/common/etcd.conf.yml '|' grep token
- Stdout: |
    # Initial cluster token for the etcd cluster during bootstrap.
    initial-cluster-token: 'FU88K7'
  UnitId: etcd/0
- Stdout: |
    # Initial cluster token for the etcd cluster during bootstrap.
    initial-cluster-token: 'FU88K7'
  UnitId: etcd/1
- Stdout: |
    # Initial cluster token for the etcd cluster during bootstrap.
    initial-cluster-token: 'FU88K7'
  UnitId: etcd/2
```

Thanks again for the catch!